### PR TITLE
Cleanup: move internal modules in subdirectories and remove/integrate util scripts

### PR DESCRIPTION
**Pull request by stdweird**
Based on #160 and #156 and new build-tools with https://github.com/quattor/maven-tools/pull/121

Solving issues due to backwards incompatibility should not require the new release, 16.10 is sufficient

### CODE REVIEW

**Comment by stdweird** [comment]
Biggest change in component code to resolve any backwards incompatible issues will be to import `unescape` from `CCM::Path` instead of `CCM::Element`.
Any other usage of CCM modules directly is probably a mistake or some legacy behaviour that can be resovled with `getTree` usage.

I'd like to get master working on all repos without any backports for incompatibility; only after that stage, i can readd e.g. `CCM::Element` with unescape exports.

File: src/main/bin/ccm-fetch [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/perl -w
+#!/usr/bin/perl ## no critic ((RequireVersionVar);
```
Comment (ned21) [inline src/main/bin/ccm-fetch]: I am not in favour of having to modify every single file with this per-line opt-out. Why not just disable the policy globally in PerlCritic? It's only activated at level 2 (the second highest) and perlcritic tends to get annoying and less useful at level 3, so I can't see us ever wanting to enable level 2 policies.

Comment (stdweird) [inline src/main/bin/ccm-fetch]: this workaround is only needed for scripts (so i think 4 or 5 in all repos); and this is a bug in critic, see https://github.com/Perl-Critic/Perl-Critic/pull/711

Comment (ned21) [inline src/main/bin/ccm-fetch]: That makes me even less keen on it - it should only be needed temporarily and no one will be brave enough to  remove it!

Comment (stdweird) [inline src/main/bin/ccm-fetch]: is resolved now

File: src/main/bin/ccm-initialise [hunk 2]
```diff
@@ -127,7 +127,11 @@ MakeCacheRoot($this_app, $cache_root, $dopts);
 
 # make global lock file (mask and GID take care of the permissions)
 $this_app->verbose("Creating lockfile");
+<<<<<<< a5c81c552251e536424fd3de83ed2560cb70e952
```
Comment (ned21) [inline src/main/bin/ccm-initialise]: Looks like you've accidentally checked in a merge error?

**Comment by ned21** [comment]
Looks like this needs a rebase against master once #160 is merged? 32ac5a0 should be squashed into 	0e8bbc0 at the same time, in which case I think both commits cancel each other out.

**Comment by stdweird** [comment]
@ned21 it's not that easy, it changes history from another PR if i do this. i'll clean this up once the other PRs are merged.

**Comment by stdweird** [comment]
@ned21 rebased

File: src/main/perl/CacheManager/Element.pm [hunk 3]
```diff
@@ -640,61 +580,34 @@ SWITCH:
...
+
 #
 # _resolve_eid($prof_dir, $ele_path)
 #
-# Private function that resolve element's id number. $prof_dir is the profile
+# Private function that resolve element's encoded id number. $prof_dir is the profile
 # full directory path, and $ele_path is the element path (as string)
 #
-sub _resolve_eid
+sub _resolve_enc_eid
```
Comment (ned21) [inline src/main/perl/CacheManager/Element.pm]: This seems to rename every instance of eid to enc_eid. However isn't eid already short for "encoded id" -- in which case this is now a double encoded id? Is the enc added for clarity? In which case, should it be _resolve_enc_id ?

**Comment by ned21** [comment]
Code looks OK except for the one question above. Now it's all rebased, is there a reason	3971e5c can't be squashed into the commit that introduced it?

File: src/main/perl/CacheManager/Element.pm (hunk 3 above)
Comment (stdweird) [inline src/main/perl/CacheManager/Element.pm]: can you reread the pod in `Encode.pm` to see if it makes any sense at all? (it should also explain what an `eid` is (and it's not "encoded id", it's probably short for `euhm, index?` which is the response to "and how should we abbreviate this path counter?" 😄 )

**Comment by stdweird** [comment]
@ned21 i have to look into the the squashing. it's a commit in the master, i'm afraid this will involve a `push force`  to master

**Comment by ned21** [comment]
I think you have a conflict with master but the conflict resolution in 10bfd63 is wrong (it results in things that are Not Perl). Sometimes git remembers previous conflict resolutions which can make it hard to redo them but if you squash the later commit into 10bfd63 then I think everything will be fine.

**Comment by stdweird** [comment]
It's now a lot less backwards incompatible. the `warn` will prevent moving beyond maven-tools 1.50

**Comment by jrha** [comment]
Discussed at March 2017 workshop, no issues with the code but the first commit adds a set of merge conflict markers, which is very ugly.

**Comment by stdweird** [comment]
@ned21 both commits deleted

File: src/main/perl/CacheManager/Element.pm (hunk 3 above)
Comment (ned21) [inline src/main/perl/CacheManager/Element.pm]: OK, pod in Encode.pm refers to "encoded eid" so I guess this is all consistent.

**Pull request merged**